### ISSUE TITLE
ansible-lint expects "localhost," file to be present if using unparsed_is_failed in ansible.cfg

### ISSUE THREAD

**Issue by damosc** [ansible/ansible-lint#3650]
##### Summary

With `unparsed_is_failed = True` in my ansible.cfg, `ansible-lint` calls `ansible-playbook -i localhost, --syntax-check`. It's probably caused by https://github.com/ansible/ansible-lint/blob/df0c46dcc53ab46156016f14fd80bebad9545bfb/src/ansiblelint/runner.py#L332

##### Issue Type

- Bug Report

##### OS / ENVIRONMENT

```console
ansible-lint --version
ansible-lint 6.17.2 using ansible-core:2.15.2 ansible-compat:4.1.2 ruamel-yaml:None ruamel-yaml-clib:None
```

- ansible installation method: pip
- ansible-lint installation method: pip

##### STEPS TO REPRODUCE

Add `unparsed_is_failed = True` to your ansible.cfg and execute any ansible-lint command.

##### Desired Behavior

Not sure what should happen, but for sure ansible-lint cannot expect localhost, (with comma) inventory file to be present.

##### Actual Behavior

```
WARNING  Skipped installing old role dependencies due to running in offline mode.
WARNING  Skipped installing collection dependencies due to running in offline mode.
WARNING  Skipped installing old role dependencies due to running in offline mode.
WARNING  Skipped installing collection dependencies due to running in offline mode.
WARNING  Listing 1 violation(s) that are fatal
internal-error: Unexpected error code 1 from execution of: ansible-playbook -i localhost, --syntax-check local.yml
local.yml:1 [WARNING]: Unable to parse localhost, as an inventory source
ERROR! No inventory was parsed, please check your configuration and options.


Read documentation for instructions on how to ignore specific rule violations.

              Rule Violation Summary               
 count tag            profile rule associated tags 
     1 internal-error min     core                 

Failed: 1 failure(s), 0 warning(s) on 1 files.
```

```[tasklist]
### Tasks
- [ ] https://github.com/ansible/ansible-lint/pull/3680
```
